### ISSUE TITLE
When gRPC (50051) port is taken, Dozer boots up silently without reporting any error but Live won't work

### ISSUE THREAD

**Issue by snork-alt** [getdozer/dozer#1981]
In cases when gRPC port 50051 is already taken, Dozer boots up without reporting any error even if it cannot bind to that port.
As a result, many functionalities of Live are broken.

When starting locally we should always find the next available port and bind to that.

**Comment by v3g42** [comment]
Allocate random ports(take the next port if not available) and pass those parts in the query string for live.

**Comment by chubei** [comment]
I don't think we should allocate random ports. It'll make live behaviour and running behaviour inconsistent.

**Comment by v3g42** [comment]
Actually you are right. Random ports won't work. But I guess we should just pick different ports for grpc and rest and not use 8080 and 50051

**Comment by chubei** [comment]
Changing default makes sense. Any suggestions?

**Comment by snork-alt** [comment]
Let's start with high unreserved port. If that post is taken, we try the next one and so on until we find the first available one

**Comment by chubei** [comment]
> Let's start with high unreserved port. If that post is taken, we try the next one and so on until we find the first available one

We also do this on dozer run?
